src/Utils/obiDialogParser.ts

```diff
@@ -71,58 +92,53 @@ export class ObiDialogParser {
-                            }
-                            else {
-                                if (step.$type === OBIStepType.BEGIN_DIALOG && typeof step.dialog === "string") {
-                                    
-                                    const subDialog =  this.composerDialog.dialogs.find(d => d.$id === step.dialog)
-                                    if (!subDialog) {
-                                        throw new Error(`Dialog name ${step.dialog} undefined`)
-                                    }
-    
-                                    const childDialogs = await this.getTrainDialogsfromOBIDialog(subDialog)
-                                    
-                                    // Add extractor step to all the children
-                                    childDialogs.forEach(td => {
-                                        td.rounds[0].extractorStep = extractorStep
-                                    })
-    
-                                    // Add children to train dialog list
-                                    trainDialogs = [...trainDialogs, ...childDialogs]
-                                }
-                                else {
-                                    console.log(`Unhandled OBI Type: ${step.$type}`)
-                                }
-                            }
+                    if (!rule.steps) {
+                        continue
+                    }
+                    for (const step of rule.steps) {
+                        if (typeof step === "string") {
+                            throw new Error("Unexpected string step")
+                        }
+                        if (step.$type !== OBIStepType.BEGIN_DIALOG || typeof step.dialog !== "string") {
+                            console.log(`Unhandled OBI Type: ${step.$type}`)
+                            continue
                         }
+                        const subDialog = this.composerDialog.dialogs.find(d => d.$id === step.dialog)
+                        if (!subDialog) {
+                            throw new Error(`Dialog name ${step.dialog} undefined`)
+                        }
+                        const childDialogs = await this.getTrainDialogsfromOBIDialog(subDialog)
+                        // Add extractor step to all the children
+                        childDialogs.forEach(td => {
+                            td.rounds[0].extractorStep = extractorStep
+                        })
+                        // Add children to train dialog list
+                        trainDialogs = [...trainDialogs, ...childDialogs]
                     }
                 }
             }
         }
         let trainRound: CLM.TrainRound | undefined
         if (obiDialog.steps) {
-            for (const step of obiDialog.steps) {
-                if (typeof step === "string") {
+            for (let i = 0; i < obiDialog.steps.length; i = i + 1) {
```

Comment:
Seems like you only changed this to get the index. Would prefer `for (const [i, step] of obiDialog.steps.entries())`.

Reply:
Nifty.  Done.